### PR TITLE
Create Weaviate interfaces to call Typhon Weaviate instance

### PR DESCRIPTION
**Pull request by slothPete7773**
Added

- POST /api/v1/ty : search for related code with query the same as Elasticsearch.
- GET /api/v1/ty/all : retrieve all of Code objects in the Weaviate.
- GET /api/v1/ty : Just a regular Get for showing cool greeting message.

Changes
- A few changes in Gitignore, but keep both branches.
- Added a new module in `package.json` .
- Recreate `package-lock.json` by delete it and reinstall dependencies.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,57 +1,115 @@
...
 
-const URI = 'http://docker40937-typhon-server.th1.proen.cloud'
-const INDEX = 'poster'
+const URI = 'http://docker40937-typhon-server.th1.proen.cloud';
+const INDEX = 'poster';
 // index list: 'pre-test', 'poster'
 
+const weaviateClient = weaviate.client({
+	scheme: 'http',
+	host: '202.151.177.149:81',
```
Comment (annibuliful) [inline index.js]: I think you should keep it as credential information in .env file or export as user data on your cloud server

File: index.js [hunk 2]
```diff
@@ -1,57 +1,115 @@
...
 
 const app = express();
 
-app.use(bodyParser.json())
+app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
 
-const URI = 'http://docker40937-typhon-server.th1.proen.cloud'
-const INDEX = 'poster'
+const URI = 'http://docker40937-typhon-server.th1.proen.cloud';
```
Comment (annibuliful) [inline index.js]: I think you should keep it as credential information in .env file or export as user data on your cloud server

File: index.js [hunk 3]
```diff
@@ -1,57 +1,115 @@
...
+		.withFields('code _additional { id }')
+		.do();
+
+	const code_results = result['data']['Get']['Code'].map((item) => ({
+		code: item['code'],
+		score: item['_additional']['certainty'],
+		id: item['_additional']['id'],
+	}));
+
+	data = {
```
Comment (annibuliful) [inline index.js]: Did you forget to declare this variable?

File: index.js [hunk 4]
```diff
@@ -1,57 +1,115 @@
...
+		.withFields('code _additional { id certainty }')
+		.do();
+
+	const code_results = fetchedResult['data']['Get']['Code'].map((item) => ({
+		code: item['code'],
+		score: item['_additional']['certainty'],
+		id: item['_additional']['id'],
+	}));
+
+	data = {
```
Comment (annibuliful) [inline index.js]: Did you forget to declare this variable?

File: index.js [hunk 5]
```diff
@@ -1,57 +1,115 @@
...
+	};
+
+	const fetchedResult = await weaviateClient.graphql
+		.get()
+		.withClassName('Code')
+		.withNearText(nearText)
+		.withFields('code _additional { id certainty }')
+		.do();
+
+	const code_results = fetchedResult['data']['Get']['Code'].map((item) => ({
```
Comment (annibuliful) [inline index.js]: Please use nullish coleasing to prevent nullable or undefined values

File: index.js [hunk 6]
```diff
@@ -1,57 +1,115 @@
...
+	}));
+
+	data = {
+		totalHits: code_results.length,
+		hits: code_results,
+	};
+	res.status(200).json(data);
+});
+
+app.get('/api/v1/ty', async (req, res) => {
```
Comment (palm11695) [inline index.js]:
```suggestion
app.get('/api/v1/ml', async (req, res) => {
```
use `/api/v1/ml` instead of `api/v1/ty`

File: index.js [hunk 7]
```diff
@@ -1,57 +1,115 @@
...
+		},
+	});
 });
 
 app.get('/api/v1/ir', (req, res) => {
-  res.status(200).json({ message: 'Typhon: Hello World' });
+	res.status(200).json({ message: 'Typhon: Hello World' });
+});
+
+app.get('/api/v1/ty/all', async (req, res) => {
```
Comment (palm11695) [inline index.js]:
```suggestion
app.get('/api/v1/ml/all', async (req, res) => {
```

File: index.js [hunk 8]
```diff
@@ -1,57 +1,115 @@
...
+	}));
+
+	data = {
+		totalHits: code_results.length,
+		hits: code_results,
+	};
+	res.status(200).json(data);
+});
+
+app.post('/api/v1/ty', async (req, res) => {
```
Comment (palm11695) [inline index.js]:
```suggestion
app.post('/api/v1/ml', async (req, res) => {
```

**Review by palm11695 — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (cragkhit) [reply index.js]: Wow! Thanks for helping. I appreciate it 😀

**Comment by palm11695** [comment]
Please handle the error response for `api/v1/ml`
just simple handling (see example -> `/api/v1/ir`)
@slothPete7773

File: index.js (hunk 1 above)
Comment (palm11695) [reply index.js]: @slothPete7773 
If you change them to .env, please provide .env.example too.

Comment (slothPete7773) [reply index.js]: Okay, I was just confuse how to inform anyone about the inside `.env`

File: index.js [hunk 9]
```diff
@@ -8,13 +10,13 @@ const app = express();
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
 
-const URI = 'http://docker40937-typhon-server.th1.proen.cloud';
+const URI = process.env.elasticSearchAaddress;
```
Comment (palm11695) [inline index.js:13]: typo ?

**Review by palm11695 — APPROVED** [review]

**Pull request merged**